src/components/settings/settings.js

```diff
@@ -178,9 +196,11 @@ class SettingsControl extends BaseComponent {
         return acc;
       }, []);
 
+
+    let heightToBandwidthMap = this.mapHeightToMaxBandwidth(props.videoTracks);
     let qualityOptions = props.videoTracks
       .filter(t => {
-        return t.bandwidth || t.height
+        return (t.bandwidth || t.height) && (t.bandwidth === heightToBandwidthMap[t.height])
       })
       .sort((a, b) => {
```

Comment:
Why not add `mapHeightToMaxBandwidth` logic as a filter pipe and remove it in a FP manner?